### PR TITLE
fix: clib-package.c memory leaks

### PR DESCRIPTION
**Pull request by Isty001**
This patch fixes leaks reported by Valgrind in clib-package.c.

```
cd test/package
make valgrind
```

### CODE REVIEW

File: src/common/clib-package.c [hunk 1]
```diff
@@ -657,6 +657,10 @@ clib_package_new_from_slug_with_package_name(const char *slug, int verbose,
 #ifdef HAVE_PTHREADS
       init_curl_share();
       _debug("GET %s", json_url);
+      if (res) {
+        // clean up when retrying
+        http_get_free(res);
+      }
```
Comment (stephenmathieson) [inline src/common/clib-package.c:663]: IIRC, `http_get_free()` has a NULL check. We can _always_ do this:

```suggestion
      // clean up when retrying
      http_get_free(res);
```

File: test/cache/cache-test.c [hunk 2]
```diff
@@ -77,7 +77,7 @@ main() {
     it("should manage the search cache") {
       char *cached_search;
 
-  clib_cache_delete_search();
+      clib_cache_delete_search();
```
Comment (stephenmathieson) [inline test/cache/cache-test.c:80]: Ah whoops, #209 should have formatted tests too. I forgot we have `.c` tests now :)

**Review by stephenmathieson — APPROVED** [review]

File: test/cache/cache-test.c (hunk 2 above)
Comment (stephenmathieson) [reply test/cache/cache-test.c:80]: #211 :)

**Review by jwerle — APPROVED** [review]

**Comment by jwerle** [comment]
Looks good from me with the exception of stephens comments!

**Pull request merged**